File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int size = nums.size();
        vector<int> sums(size);
        int largest_sum =  sums[0] = nums[0];
        for (int i = 1; i < size; i++) {
            largest_sum = max(largest_sum, sums[i] = max(nums[i], sums[i-1] + nums[i]));
        }
        return largest_sum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int len = nums.length;
        int currentSum = nums[0];
        int maxSum = nums[0];

        for (int i = 1; i < len; i++) {
            currentSum = Math.max(nums[i], currentSum + nums[i]);
            maxSum = Math.max(maxSum, currentSum);
        }

        return maxSum;
    }
}